holoviews/operation/datashader.py

```diff
@@ -319,19 +312,94 @@ def _process(self, element, key=None):
         params = dict(get_param_values(element), kdims=element.dimensions()[:2],
                       datatype=['xarray'], vdims=vdims)
 
-        agg = getattr(cvs, glyph)(data, x, y, self.p.aggregator)
+        dfdata = PandasInterface.as_dframe(data)
+        agg = getattr(cvs, glyph)(dfdata, x, y, self.p.aggregator)
+        if 'x_axis' in agg and 'y_axis' in agg:
+            agg = agg.rename({'x_axis': x, 'y_axis': y})
+
         if agg.ndim == 2:
             # Replacing x and y coordinates to avoid numerical precision issues
-            return self.p.element_type((xs, ys, agg.data), **params)
+            eldata = agg if ds_version > '0.5.0' else (xs, ys, agg.data)
+            return self.p.element_type(eldata, **params)
         else:
             layers = {}
             for c in agg.coords[column].data:
                 cagg = agg.sel(**{column: c})
-                layers[c] = self.p.element_type((xs, ys, cagg.data), **params)
+                eldata = cagg if ds_version > '0.5.0' else (xs, ys, cagg.data)
+                layers[c] = self.p.element_type(eldata, **params)
             return NdOverlay(layers, kdims=[data.get_dimension(column)])
 
 
 
+class regrid(resample_operation):
+    """
+    Regridding allows resampling a HoloViews Image type using specified
+    up- and downsampling functions.
+    """
+
+    aggregator = param.ObjectSelector(default='mean',
+        objects=['first', 'last', 'mean', 'mode', 'std', 'var', 'min', 'max'], doc="""
+        Aggregation method.
+        """)
+
+    interpolation = param.ObjectSelector(default='nearest',
+        objects=['linear', 'nearest'], doc="""
+        Interpolation method""")
+
+    upsample = param.Boolean(default=True, doc="""
+        Whether to allow upsampling if the source array is smaller than
+        the requested array.""")
```

Comment:
Should this be False by default since interpolation is nearest by default, and thus the result will be approximately the same whether or not upsampling is done?  With linear interpolation the results will be different, but one could say in the docstring for the interpolation parameter that one might want upsample=False for interpolation methods that smooth the data, unlike nearest.